src/datasource/OpenSearchResponse.ts

```diff
@@ -666,31 +666,38 @@ type Doc = {
   _id: string;
   _type: string;
   _index: string;
+  _score?: any;
   _source?: any;
 };
 
 /**
  * Flatten the docs from response mainly the _source part which can be nested. This flattens it so that it is one level
  * deep and the keys are: `level1Name.level2Name...`. Also returns list of all properties from all the docs (not all
  * docs have to have the same keys).
- * @param hits
+ * @param hits - The hits from KalDB
+ * @param timeField - The time field set in the datasource so that we can extract it from the hits payload
  */
-const flattenHits = (hits: Doc[]): { docs: Array<Record<string, any>>; propNames: string[] } => {
+const flattenHits = (hits: Doc[], timeField: string): { docs: Array<Record<string, any>>; propNames: string[] } => {
   const docs: any[] = [];
   // We keep a list of all props so that we can create all the fields in the dataFrame, this can lead
   // to wide sparse dataframes in case the scheme is different per document.
   let propNames: string[] = [];
 
   for (const hit of hits) {
     const flattened = hit._source ? flatten(hit._source) : {};
-    const doc = {
+
+    let doc = {
       _id: hit._id,
       _type: hit._type,
       _index: hit._index,
+      _score: hit._score,
       _source: { ...flattened },
       ...flattened,
     };
 
+    // The hit.<timeField> field is in fractional seconds, but we expect it to be in milliseconds down the line
+    doc[timeField] = hit[timeField] * 1000;
```

Comment:
Did we consider an approach like https://github.com/grafana/opensearch-datasource/pull/66? 

Looking at the changes to the upstream file here - https://github.com/grafana/opensearch-datasource/commits/main/src/OpenSearchResponse.ts, there's not been a ton of updates they did for Grafana 9 support. As far as I can tell they just now look at the field type and automatically convert all date types, which may be how they got around needing to pass the timefield name?